pkgs/by-name/pe/perfect_dark/package.nix

```
@@ -0,0 +1,120 @@
+{
+  romID ? "ntsc-final",
+
+  lib,
+
+  fetchFromGitHub,
+  stdenv_32bit,
+
+  pkgsi686Linux, # SDL2 and zlib
+
+  pkg-config,
+  python3,
+}:
+let
+  roms = {
+    "ntsc-final" = {
+      binName = "pd";
+    };
+    "pal-final" = {
+      binName = "pd.pal";
+    };
+    "jpn-final" = {
+      binName = "pd.jpn";
+    };
+  };
+  binName = roms.${romID}.binName;
+in
+assert lib.assertOneOf "romID" romID (builtins.attrNames roms);
+
+stdenv_32bit.mkDerivation rec {
+  pname = "perfect_dark";
+  version = "0-unstable-2024-09-02";
+
+  src = fetchFromGitHub {
+    owner = "fgsfdsfgs";
+    repo = "perfect_dark";
+    rev = "2a5c3a351eeb1772306567969fb8dc5b31eaf34e";
+    hash = "sha256-tpAzpIe2NYUtIY3NsvGl9liOuNb4YQCcfs+oLkFpFQA=";
+  };
+
+  buildInputs = with pkgsi686Linux; [
+    SDL2
+    zlib
+  ];
+
+  nativeBuildInputs = [
+    pkg-config
+    python3
```

```suggestion
    python3Packages.python
```
Minor nit, as this probably won't need to run on other Python versions.